__tests__/components/MnemonicInput.spec.ts

```diff
@@ -13,53 +13,482 @@
+                wordToBeDeleted: boolean;
+            },
+        ) => {
+            // Arrange:
+            const props = {
+                seed: 'word',
+            };
+            const mockInitInput = jest.fn();
+            const mockHandleWordsArray = jest.fn();
+
+            // Act:
+            const wrapper = getMnemonicInputWrapper(props);
+            const component = wrapper.vm as MnemonicInputTs;
+            component.initInput = mockInitInput;
+            component.handleWordsArray = mockHandleWordsArray;
+            component.inputWord = inputWord;
+            component.isEditing = isEditing;
+            component.isNeedPressDelTwice = isNeedPressDelTwice;
+            component.deleteWord();
+
+            // Assert:
+            if (expectations.wordToBeDeleted) {
+                expect(mockInitInput).toBeCalled();
+                expect(mockHandleWordsArray).toBeCalledWith();
+            } else {
+                expect(mockInitInput).not.toBeCalled();
+                expect(mockHandleWordsArray).not.toBeCalled();
+            }
+            expect(component.isNeedPressDelTwice).toBe(expectations.finalIsNeedPressDelTwice);
+        };
 
-        // assert
-        expect(vm.showCopyButton).toBe(true);
-        expect(wrapper.findComponent(ButtonCopyToClipboard).exists()).toBe(true);
-        expect(wrapper.find('.paste').exists()).toBe(false);
-        wrapper.destroy();
+        test('not delete word when inputWord is not empty', () => {
+            // Arrange:
+            const inputWord = 'word';
+            const isEditing = false;
+            const isNeedPressDelTwice = false;
+            const expectations = {
+                finalIsNeedPressDelTwice: true,
+                wordToBeDeleted: false,
+            };
+
+            // Act + Assert:
+            runAddWordTest(inputWord, isEditing, isNeedPressDelTwice, expectations);
+        });
+
+        test('not delete word when isNeedPressDelTwice is true', () => {
+            // Arrange:
+            const inputWord = '';
+            const isEditing = true;
+            const isNeedPressDelTwice = true;
+            const expectations = {
+                finalIsNeedPressDelTwice: false,
+                wordToBeDeleted: false,
+            };
+
+            // Act + Assert:
+            runAddWordTest(inputWord, isEditing, isNeedPressDelTwice, expectations);
+        });
+
+        test('delete word when inputWord is empty', () => {
+            // Arrange:
+            const inputWord = '';
+            const isEditing = false;
+            const isNeedPressDelTwice = false;
+            const expectations = {
+                finalIsNeedPressDelTwice: false,
+                wordToBeDeleted: true,
+            };
+
+            // Act + Assert:
+            runAddWordTest(inputWord, isEditing, isNeedPressDelTwice, expectations);
+        });
+
+        test('delete word when inputWord is empty and isEditing is true', () => {
```

Comment:
can refactor as this test is same as previous one, with only difference being isEditing value